apps/mobile/src/components/TradeChecklistFlow/components/CalculateAmountFlow/components/FiatAmountInput.tsx

```diff
@@ -0,0 +1,105 @@
+import {useCallback, useRef, useState} from 'react'
+import {type TextInput} from 'react-native'
+import {type PrimitiveAtom, useAtom, useAtomValue, useSetAtom} from 'jotai'
+import {
+  btcOrSatAtom,
+  feeAmountAtom,
+  SATOSHIS_IN_BTC,
+  tradeBtcPriceAtom,
+} from '../atoms'
+import {checkInputValueIsDecimal} from '../../../utils'
+import {Stack, Text} from 'tamagui'
+import AmountInput from './AmountInput'
+import {currencies} from '../../../../../utils/localization/currency'
+import {offerForTradeChecklistAtom} from '../../../atoms'
+import CalculatedWithLiveRate from './CalculatedWithLiveRate'
+
+interface Props {
+  automaticCalculationDisabled?: boolean
+  fiatValueAtom: PrimitiveAtom<string | undefined>
+  btcValueAtom: PrimitiveAtom<string | undefined>
+  showSubtitle?: boolean
+}
+
+function FiatAmountInput({
+  automaticCalculationDisabled,
+  btcValueAtom,
+  fiatValueAtom,
+  showSubtitle,
+}: Props): JSX.Element {
+  const ref = useRef<TextInput>(null)
+  const [isFocused, setIsFocused] = useState<boolean>(false)
+
+  const [fiatValue, setFiatValue] = useAtom(fiatValueAtom)
+  const setBtcValue = useSetAtom(btcValueAtom)
+  const feeAmount = useAtomValue(feeAmountAtom)
+  const offerForTradeChecklist = useAtomValue(offerForTradeChecklistAtom)
+  const tradeBtcPrice = useAtomValue(tradeBtcPriceAtom)
+  const btcOrSat = useAtomValue(btcOrSatAtom)
+
+  const calculateBtcValueOnFiatAmountChange = useCallback(
+    (value: string | undefined) => {
+      setFiatValue(value)
+      if (!automaticCalculationDisabled) {
+        if (tradeBtcPrice && tradeBtcPrice !== 0 && value) {
+          const numberValue = Number(value)
+          const fee = (numberValue / tradeBtcPrice) * (feeAmount / 100)
+          setBtcValue(
+            btcOrSat === 'BTC'
+              ? (numberValue / tradeBtcPrice - fee).toFixed(6)
+              : `${Math.round(
+                  (numberValue / tradeBtcPrice - fee) * SATOSHIS_IN_BTC
+                )}`
+          )
+        } else {
+          setBtcValue(undefined)
+        }
+      }
+    },
+    [
+      automaticCalculationDisabled,
+      btcOrSat,
+      feeAmount,
+      setBtcValue,
+      setFiatValue,
+      tradeBtcPrice,
+    ]
+  )
+
+  return (
+    <AmountInput
+      ref={ref}
+      showSubtitle={showSubtitle}
+      isFocused={isFocused}
+      onBlur={() => {
+        setIsFocused(false)
+      }}
+      onFocus={() => {
+        setIsFocused(true)
+      }}
+      onWrapperPress={() => {
+        ref.current?.focus()
+      }}
+      placeholder={`${offerForTradeChecklist?.offerInfo.publicPart.amountTopLimit}`}
+      value={fiatValue}
+      onChangeText={(input) => {
+        calculateBtcValueOnFiatAmountChange(checkInputValueIsDecimal(input))
```

Comment:
Is it ok to do this on every key press?

Reply:
I had problems with debouncing this value as btcInput changes the value of fiatInput and also fiatInput changes the value of btcInput. So if fiatInput changes, then it changes the btcInput and on that change another change of fiatInput is triggered. Cause with debouncing I need to rely on some useEffect calling the calculation function. Maybe if u have some better solution I am open to implement it, maybe I just don't see it correctly.

Reply:
I don't understand how adding debouncing would result in triggering changes that are not triggered by updates without debounce.

I am afraid, that just user typing will cause extensive rerenders but it also makes sense to update btcPriceInput when FiatInput is changed (and vice versa). So let's just make sure that calling this method does not trigger unnecessary rerenders and we're golden!